core/src/main/java/org/bitcoinj/crypto/HDPath.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2019 Michael Sean Gilligan.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bitcoinj.crypto;
+
+import java.util.AbstractList;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * HD Key derivation path. {@code HDPath} can be used to represent a full path or a relative path.
+ * The {@code hasPrivateKey} {@code boolean} is used for rendering to {@code String}
+ * but (at present) not much else. It defaults to {@code false} which will also be the setting for a relative path.
+ * <p>
+ * {@code HDPath} is immutable and uses the {@code Collections.UnmodifiableList} type internally.
+ * <p>
+ * It implements {@code java.util.List<ChildNumber>} to ease migration
+ * from the previous Guava {@code ImmutableList<ChildNumber>}. It should be a minor breaking change
+ * to replace {@code ImmutableList<ChildNumber>} with {@code List<ChildNumber>} where necessary in your code. Although
+ * it is recommended to use the {@code HDPath} type for clarity and for access to {@code HDPath}-specific functionality.
+ */
+public class HDPath extends AbstractList<ChildNumber> {
+    static final char PREFIX_PRIVATE = 'm';
```

Comment:
Can these constants be private?

Reply:
They can, but it seems like they should be made available as public constants _somewhere_.

Reply:
Somehow I can't think of a reason to have them public. The textual representation of a path is owned by HDPath, no?

Reply:
I can think of a couple of at least one use case: parsing a path string. I don't think we have code to do that. And if even if we did have that code, we're using `44H` and someone might want to implement the `44'` format.

Reply:
That said, I'm happy to make them private if you prefer.

Reply:
I think yes, for now I'd prefer to have them private.

Reply:
Fixed in force-push.